### ISSUE TITLE
Async versions of given/when/then don't catch panic

### ISSUE THREAD

**Issue by nbigaouette** [bbqsrc/cucumber-rust#66]
With 0.7 async steps are supported (thanks!) but if they panic the binary exists with return code of 0 (meaning success). Cucumber does not show the failure nor any final output, it simply exits.

```rust
steps
        .given_async("Panic async (caught)", 
        Rc::new(|world, _step| {
            std::panic::AssertUnwindSafe(async move { 
                println!("Panic async!!!");
                unimplemented!();
            })
            .catch_unwind()
            .boxed_local()
        }))
        .given_async("Panic async (problematic)", 
        Rc::new(|world, _step| {
            async move { 
                println!("Panic async!!!");
                unimplemented!();
            })
        }))
```

While the `Panic async (caught)` step is properly caught and show the error, the `Panic async (problematic)` will terminate the binary without error nor output.

It would be helpful for users to automatically adjust the closure to be wrapped in both the `Rc` and the `AssertUnwindSafe` so that both the sync and async versions are easy to use.

**Comment by bbqsrc** [comment]
Not a bug, or at least not something I can fix due to how async/await works (based on my understanding). Just literally how this works if you don't box your functions with catch_unwind. I added the `t!` macro to simplify declaring async functions, which should be visible in the example in the readme, though I suppose I should call this out in more detail.

Would you be interested in adding an FAQ section that includes the problem you just hit and the solution? :smile:

**Comment by bbqsrc** [comment]
Actually I'm having a dumb day, I think I just forgot a `+ UnwindSafe`.

**Comment by bbqsrc** [comment]
Yeah it was that. Fixed in 0.7.2. :smile:
